### PR TITLE
SCC - Action Input Form 

### PR DESCRIPTION
**Pull request by RakeshUP**
- Only two data types have been added to the action input form
- Links the SCC to the new proposal page

Task: [APP-2138](https://aragonassociation.atlassian.net/browse/APP-2138), [APP-1253](https://aragonassociation.atlassian.net/browse/APP-1253)

Use addresses from [https://goerli.etherscan.io/contractsVerified](https://goerli.etherscan.io/contractsVerified) to test contract validation and actions. 

Here's a list of contracts that I used
0x24610BA80a89aAdf4FaF7A9944D00C33080E786b

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x]  I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-2138]: https://aragonassociation.atlassian.net/browse/APP-2138?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[APP-1253]: https://aragonassociation.atlassian.net/browse/APP-1253?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jamesej — COMMENTED** [review]
That looks really great, it's pretty concise. One thing though, you'll need to cover all the possible primitive types like `fixed` etc [https://docs.soliditylang.org/en/v0.8.19/abi-spec.html#types](https://docs.soliditylang.org/en/v0.8.19/abi-spec.html#types)

File: packages/web-app/src/containers/smartContractComposer/index.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  useEffect(() => {
+    if (address) {
+      const storedContracts = getVerifiedSmartContracts(
+        address,
+        CHAIN_METADATA[network].id
+      );
+
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```
Comment (sepehr2github) [inline packages/web-app/src/containers/smartContractComposer/index.tsx:51]: Why did you ignore ts?

**Review by sepehr2github — CHANGES_REQUESTED** [review]
The modal will remove the empty state when I close it 
Screenshot 2023-04-27 at 12 39 53 PM

Also if I jump back then Jump next the modal pops up

File: packages/web-app/src/containers/smartContractComposer/components/inputForm.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+            />
+          )}
+        />
+      );
+
+    case 'tuple':
+      console.log(input.type);
+      input.components?.map(component => (
+        <div key={component.name}>
+          <div className="mb-1.5 text-base font-bold capitalize text-ui-800">
```
Comment (sepehr2github) [inline packages/web-app/src/containers/smartContractComposer/components/inputForm.tsx]: Lint!

File: packages/web-app/src/containers/smartContractComposer/components/inputForm.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      </p>
+      <p className="mt-1 text-sm text-ui-600">
+        @notice as help text; This is the description of the method provided by
+        NatSpec Format or if those are our smart contracts, by further
+        implementation
+      </p>
+      <div className="p-3 mt-5 space-y-2 rounded-xl bg-ui-50 border-ui-100 shadow-100">
+        {selectedAction.inputs.map(input => (
+          <div key={input.name}>
+            <div className="mb-1.5 text-base font-bold capitalize text-ui-800">
```
Comment (sepehr2github) [inline packages/web-app/src/containers/smartContractComposer/components/inputForm.tsx]: Please fix all lint errors

File: packages/web-app/src/containers/smartContractComposer/index.tsx (hunk 1 above)
Comment (RakeshUP) [reply packages/web-app/src/containers/smartContractComposer/index.tsx:51]: There's a [cyclic type dependency here](https://github.com/aragon/app/blob/13b77d9eaca4e1b5724e5e91ab673f71865f9789/packages/web-app/src/utils/types.ts#L386), and TS won't stop complaining about it. I've used ts-ignore on multiple places, I'll try to remove that and fix the ts issue separately

**Comment by RakeshUP** [comment]
@sepehr2github, I've fixed the lint warnings and the issues with empty state and the SCC modal. 

I'll fix the @ts-ignore thing in my upcoming PR

**Review by sepehr2github — CHANGES_REQUESTED** [review]
Thanks for the changes, One small issue still left, when I add a new contract and click to add another one, the modal input is disabled

**Review by sepehr2github — APPROVED** [review]
Thanks, will fix the issue in upcoming prs

**Pull request merged**